saleor/graphql/attribute/utils.py

```diff
@@ -188,16 +190,38 @@ def _pre_save_boolean_values(
         attr_values: AttrValuesInput,
     ):
         get_or_create = attribute.values.get_or_create
+        boolean = bool(attr_values.boolean)
         value, _ = get_or_create(
             attribute=attribute,
-            slug=slugify(f"{attribute.id}_{attr_values.boolean}", allow_unicode=True),
+            slug=slugify(f"{attribute.id}_{boolean}", allow_unicode=True),
             defaults={
-                "name": f"{attribute.name}: {'Yes' if attr_values.boolean else 'No'}",
-                "boolean": attr_values.boolean,
+                "name": f"{attribute.name}: {'Yes' if boolean else 'No'}",
+                "boolean": boolean,
```

Comment:
I think those changes are not related to Date & DateTime attributes. We should add them in separate PR. 
If we add changes not related to PR we can easily miss that we should add more tests for this bugfix. Our QA can miss that changes and don't perform required tests.